### ISSUE TITLE
Crash in nightly from 2022-07-12

### ISSUE THREAD

**Issue by cricalix** [wez/wezterm#2257]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

20220711-212031-98fb67c0

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Dragging wezterm from one display to another causes the program to die as it crosses the boundary on to the second screen. It sort of appears on the second screen, but does a shrink-to-vanish and stops running.

https://user-images.githubusercontent.com/1478917/178689229-d39517d2-1a5c-4723-af59-fd7955da20d6.mp4

### To Reproduce

Windows 10.
Twin displays - 1 @ 2560x1440, 1 @ 3840x2160. 1440 scaled at 100%, 2160 scaled at 150%.
Program starts on the 2160 display.
Drag from 2160 to 1440 or use the Windows native hotkeys (win + arrow) to send the window across to the second display.

This behaviour does not reproduce in 20220624-141144-bd1b7c5d, so this is a regression in the nightly builds.

### Configuration

```
local wezterm = require 'wezterm';

return {
  color_scheme = "Dark+",
  scrollback_lines = 3500,
  enable_scroll_bar = true,
  use_fancy_tab_bar = false,
  font = wezterm.font("Cascadia Mono"),
  font_size = 13,
  launch_menu = {
    { an ssh entry }
    },
  },
}
```

### Expected Behavior

Nightly doesn't exit when being moved from one display to another.

### Logs

Cannot capture logs - ctrl-shift-l captures into the display of wezterm-gui, and that's the process that dies.

Launching just 'wezterm' from the CLI crashes on start; have to use wezterm-gui. However, the crash on start is:
```
09:31:49.865  ←[32mINFO  ←[0m ←[1mwezterm_gui::termwindow←[0m > OpenGL initialized! Intel(R) UHD Graphics 4.5.0 - Build 30.0.101.1340 is_context_loss_possible=true wezterm version: 20220711-212031-98fb67c0
09:31:49.985  ←[31mERROR ←[0m ←[1menv_bootstrap          ←[0m > panic at window\src\os\windows\window.rs:980:23 - !?
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: CallWindowProcW
  12: CallWindowProcW
  13: glPushClientAttrib
  14: CallWindowProcW
  15: DispatchMessageW
  16: TranslateAcceleratorW
  17: KiUserCallbackDispatcher
  18: NtUserSetWindowPos
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: BaseThreadInitThunk
  30: RtlUserThreadStart

thread 'main' panicked at 'already borrowed: BorrowMutError', window\src\os\windows\window.rs:980:23
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
09:31:50.000  ←[31mERROR ←[0m ←[1mwindow::os::windows::window←[0m > caught Any { .. }
```

Non-nightly does not crash when run as 'wezterm' from a cmd prompt.

**Comment by wez** [comment]
This should be fixed now in `main`.

It typically takes about an hour before fixes are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out. You can find the nightly downloads for your system in the [wezterm installation docs](https://wezfurlong.org/wezterm/installation.html).

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them. 

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Comment by cricalix** [comment]
Confirmed fixed in 20220714-001416-810512c2
